runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -366,20 +419,26 @@ public void prefixTrim(final Token address) {
                                     });
                             return null;
                 }, true);
-                // TODO(Maithem): trimCache should be epoch aware?
-                runtime.getSequencerView().trimCache(address.getSequence());
                 break;
             } catch (NetworkException | TimeoutException e) {
                 log.warn("prefixTrim: encountered a network error on try {}", x, e);
                 Duration retryRate = runtime.getParameters().getConnectionRetryRate();
                 Sleep.sleepUninterruptibly(retryRate);
             } catch (WrongEpochException wee) {
                 long serverEpoch = wee.getCorrectEpoch();
-                long runtimeEpoch = runtime.getLayoutView().getLayout().getEpoch();
-                log.warn("prefixTrim[{}]: wrongEpochException, runtime is in epoch {}, while server is in epoch {}. "
-                                + "Invalidate layout for this client and retry, attempt: {}/{}",
-                        address, runtimeEpoch, serverEpoch, x + 1, numRetries);
-                runtime.invalidateLayout();
+                // Retry if wrongEpochException corresponds to message epoch (only)
+                if (address.getEpoch() == serverEpoch) {
+                    long runtimeEpoch = runtime.getLayoutView().getLayout().getEpoch();
+                    log.warn("prefixTrim[{}]: wrongEpochException, runtime is in epoch {}, " +
+                            "while server is in epoch {}. Invalidate layout for this client " +
+                            "and retry, attempt: {}/{}", address, runtimeEpoch, serverEpoch, x+1, numRetries);
+                    runtime.invalidateLayout();
+                } else {
+                    // wrongEpochException corresponds to a stale trim address (prefix trim token on the wrong epoch)
```

Comment:
Can you remove this? I had removed this in #1775

Reply:
Done.